packages/design-system/src/components/Dialog/Dialog.tsx

```diff
@@ -334,86 +148,90 @@ export const Dialog = (props: DialogProps) => {
     };
   }, []);
 
-  const {
-    actions,
-    actionsClassName,
-    analytics,
-    ariaCloseLabel,
-    children,
-    className,
-    closeButtonSize,
-    closeButtonText,
-    closeButtonVariation,
-    closeIcon,
-    closeText,
-    escapeExits,
-    escapeExitDisabled,
-    headerClassName,
-    heading,
-    onExit,
-    size,
-    title,
-    ...modalProps
-  } = props;
+  // Toggle dialog open
+  const dialogRef = useRef(null);
+  const lastActiveElement = React.useRef(null);
+  const firstRender = React.useRef(true);
+
+  useLayoutEffect(() => {
```

Comment:
Using `useLayoutEffect` because I think this is modifying the DOM. Still a little unclear on the difference between this hook and `useEffect`